src/components/sites-sync-modal-selector.tsx

```diff
@@ -0,0 +1,215 @@
+	syncSites,
+	selectedSiteId,
+	onSelectSite,
+}: {
+	syncSites: SyncSite[];
+	selectedSiteId: null | number;
+	onSelectSite: ( id: number ) => void;
+} ) {
+	return (
+		<div className="flex flex-col overflow-y-auto h-full">
+			{ syncSites.map( ( site ) => (
+				<SiteItem
+					key={ site.id }
+					site={ site }
+					isSelected={ site.id === selectedSiteId }
+					onClick={ () => onSelectSite( site.id ) }
+				/>
+			) ) }
+		</div>
+	);
+}
+
+function SiteItem( {
+	site,
+	isSelected,
+	onClick,
+}: {
+	site: SyncSite;
+	isSelected: boolean;
+	onClick: () => void;
+} ) {
+	const { __ } = useI18n();
+	if ( site.isStaging ) {
+		return null;
+	}
+	const isSyncable = site.syncSupport === 'syncable';
+	const isNeedsTransfer = site.syncSupport === 'needs-transfer';
+	const isUnsupported = site.syncSupport === 'unsupported';
+	return (
+		<div
+			className={ cx(
+				'flex py-3 px-8 items-center border-b border-a8c-gray-0 justify-between',
+				isSelected && 'bg-a8c-blueberry text-white',
+				! isSelected && isSyncable && 'hover:bg-a8c-blueberry-5'
+			) }
+			role={ isSyncable ? 'button' : undefined }
+			onClick={ () => {
+				if ( ! isSyncable ) {
+					return;
+				}
+				onClick();
+			} }
+		>
+			<div className="flex flex-col gap-0.5">
+				<div className={ cx( 'a8c-body', ! isSyncable && 'text-a8c-gray-30' ) }>{ site.name }</div>
+				<div className={ cx( 'a8c-body-small text-a8c-gray-30', isSelected && 'text-white' ) }>
+					{ site.url }
+				</div>
+			</div>
+			{ isSyncable && (
+				<div className="flex gap-2">
+					<Badge
+						className={ cx(
+							isSelected
+								? 'bg-white text-a8c-blueberry text-a8c-blueberry'
+								: 'bg-a8c-green-5 text-a8c-green-80'
+						) }
+					>
+						{ __( 'Production' ) }
+					</Badge>
+					{ site.stagingSiteIds.length > 0 && (
+						<Badge
+							className={ cx( isSelected && 'bg-white text-a8c-blueberry text-a8c-blueberry' ) }
+						>
+							{ __( 'Staging' ) }
+						</Badge>
+					) }
+				</div>
+			) }
+			{ isUnsupported && (
+				<div className="a8c-body-small text-a8c-gray-30"> { __( 'Unsupported plan' ) }</div>
+			) }
+			{ isNeedsTransfer && (
+				<div className="a8c-body-small text-a8c-gray-30">
+					{ __( 'Please enable hosting features' ) }
+				</div>
+			) }
+		</div>
+	);
+}
+
+function Footer( {
+	onRequestClose,
+	onConnect,
+	disabled,
+}: {
+	onRequestClose: () => void;
+	onConnect: () => void;
+	disabled: boolean;
+} ) {
+	const { __ } = useI18n();
+	return (
+		<div className="flex px-8 py-4 border-t border-a8c-gray-5 justify-between">
+			<Button
+				variant="link"
+				className="flex items-center mb-1"
+				onClick={ () => getIpcApi().openURL( 'https://wordpress.com/hosting/' ) }
+			>
+				<div className="a8c-subtitle text-black">{ __( 'Powered by' ) }</div>
+				<WordPressShortLogo className="ml-2 h-5" />
+			</Button>
+			<div className="flex gap-4">
+				<Button variant="link" onClick={ onRequestClose }>
+					{ __( 'Cancel' ) }
```

Comment:
I can also see that this button almost blends with wp.com logo when Studio window is sized to the smallest size and the modal is open:

<img width="463" alt="Screenshot 2024-10-18 at 9 55 12 AM" src="https://github.com/user-attachments/assets/f1bb12cb-5ebb-43af-be95-eb2df3543f3f">

I am wondering if we can increase the min-width of the modal in general to resolve some of these issues?

Reply:
> I am wondering if we can increase the min-width of the modal in general to resolve some of these issues?

I agree. We appear to have a lot of dead space to the left and right of the modal at this window size.

Reply:
Good catch testing the modal in a small window 👍 .

Added a minimum width, and all our problems are solved ;)  https://github.com/Automattic/studio/pull/601/commits/a960da5bcf7583143f52dcd51a1d1ead9cda160b

<img width="764" alt="Screenshot 2024-10-18 at 11 42 23" src="https://github.com/user-attachments/assets/e1b824e8-f229-4971-9789-daa774c6d0bd">